modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb

```diff
@@ -36,14 +35,43 @@ def submit
           render json: auto_claim
         end
 
-        def validate; end
+        def validate
+          shared_validation
+
+          ClaimsApi::AutoEstablishedClaim.new(
+            status: ClaimsApi::AutoEstablishedClaim::PENDING,
+            auth_headers:,
+            form_data: form_attributes,
+            cid: token.payload['cid'],
+            veteran_icn: target_veteran.mpi.icn
+          )
+          # evss_claim = evss_service.valid(auto_claim, evss_data) ???
+
+          render json: valid_526_response
+        end
 
         def attachments; end
 
         def get_pdf
           # Returns filled out 526EZ form as PDF
         end
 
+        def shared_validation
+          validate_json_schema
+          validate_form_526_submission_values!
+        end
```

Comment:
Would it be worth it to utilize a before_action with these validations?

Reply:
possibility. we'll want to look at the ordering of the before_actions but we could do something along those lines. I think it's going to be pretty consistent across all of them. (submit, attachments, validate) and we can always edit exactly which ones we want it on if there are some that aren't included. I'd wanna make sure we run our auth action first before we even waste time on the validate, which is my only concern with adding it as a before action as well, but I'm sure we can find ways to edit the order if needed.